### PR TITLE
Julian dev

### LINKED ISSUE #58: View/Template for the search survey result (currently JSON)

### LINKED ISSUE #59: Actual storage of survey answers.

### LINKED ISSUE #60: Survey Results Page

### PR DESCRIPTION
**Pull request by JulianMendoza**
- Fixes/changes to validation upon creating a new survey.
- Refactored survey submission completely so that all inputs are added to the survey
- Created a view for the survey results
- Added feature to remove surveys
- Implemented integration tests for all new features added

- More testing is required I'm sure the code will break if you try to break it

- sudo inject a survey upon visiting http://localhost:8080/
- View the results by searching for survey with code "reserved" and password "reserved" 

Current view

image

### CODE REVIEW

**Comment by loisaleghe** [comment]
When you say more testing is required, what are you referring to that may break the code and is it something we should be worried about?

**Comment by JulianMendoza** [comment]
> When you say more testing is required, what are you referring to that may break the code and is it something we should be worried about?

As a developer, you're not really sure how the end user will use the page. I haven't vigorously tested the pages myself with bad inputs. I am also not too familiar with JavaScript and page loading -> ensuring that all scripts load before the view.  For the scope of this project, I don't think it's really something to worry about since this project is more about the process of development.

**Comment by loisaleghe** [comment]
> > When you say more testing is required, what are you referring to that may break the code and is it something we should be worried about?
> 
> As a developer, you're not really sure how the end user will use the page. I haven't vigorously tested the pages myself with bad inputs. I am also not too familiar with JavaScript and page loading -> ensuring that all scripts load before the view. For the scope of this project, I don't think it's really something to worry about since this project is more about the process of development.

That is reasonable, thank you for your reply. Overall the code looks good to me

File: src/main/java/com/surveyMonkey/controllers/AdminController.java [hunk 1]
```diff
@@ -64,13 +62,18 @@ public String showSurvey(@PathVariable String surveyCode, Model model) {
...
+        for(Survey survey:surveyRepository.findAll()){
+            if(survey.getSurveyCode().equals(surveyCode)&&survey.getSurveyPassword().equals(surveyPassword)){
+                model.addAttribute("surveyCode", surveyCode);
+                break;
+            }
+        }
         return "results";
     }
 
     @PostMapping({"/surveyResult"})
```
Comment (oyenmwen) [inline src/main/java/com/surveyMonkey/controllers/AdminController.java:74]: Maybe change the name of this postmapping to be more informative or different than the previous one (/surveyResult vs /surveyResults) can be easy to get confused and make mistakes. Not that serious though.

**Review by oyenmwen — APPROVED** [review]
Great job dude. Awesome work

**Pull request merged**